build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/CommonBuilder.java

```diff
@@ -505,6 +506,8 @@ private IProject[] build(int kind, IProject project, IBuilder[] builders, boolea
 			}
 
 			for (int i = 0; i < num; i++) {
+				CompilationDatabaseGenerator generator = new CompilationDatabaseGenerator(getProject(), activeCfg);
```

Comment:
@jonahgraham my thinking was based on how the compilation databases are consumed. Ideally, the editing experience that MBS users have grown accustomed to would not regress in any way. It seems that database generation that occurs only on build would be particularly noticeable immediately following project creation or project import. I am OK with hooking into the CDT builder in the first instance with a view to assessing performance and the viability of alternative hooks. I wonder if there is performance to be gained by observing the resource delta.

Regardless, this contribution is good news for CDT. I'm looking forward to seeing the database generator implementation. 👍

Reply:
> It seems that database generation that occurs only on build would be particularly noticeable immediately following project creation or project import.

Yes, I agree. The CMake flow suffers from this - i.e. immediately on project creation you end up with a project full of red underlines/problem markers.

Thank you for supporting deferring when/where this gets hooked in. Perhaps we can/should fix both the CMake and managed build at the same time by improving the project/import wizard. The other option is improving cdt-lsp to identify when compile_commands.json is missing and prompt to do a build that will then generate it?

@alicetrifu can you add this to a TODO list in #689?

Reply:
Would be good to look at the CMake timing too. There are also other features that would be useful o have that would benefit from a CMake configure being run in the background rather than only on build.